packages/contracts/src/Wallet.sol

```diff
@@ -120,13 +128,81 @@ contract Wallet is TokenCallbackHandler, OwnableUpgradeable, UUPSUpgradeable {
         return keccak256(abi.encode(txDataTmp));
     }
 
-    function _execute(address target, uint256 value, bytes calldata data) internal {
-        (bool success, bytes memory result) = target.call{value: value}(data);
+    function _checkAllowanceReduction(bytes calldata data) internal view returns (bool) {
+        bool result = false;
+
+        bytes4 functionSelector = bytes4(data[:4]);
+        if (functionSelector == bytes4(keccak256("transfer(address,uint256)"))) {
+            result = true;
+        } else if(functionSelector == bytes4(keccak256("approve(address,uint256)"))) {
+            result = true;
+        } else if(functionSelector == bytes4(keccak256("transferFrom(address,address,uint256)"))) {
+            (address from, , ) = abi.decode(data[4:], (address, address, uint256));
+            if(from == address(this)) {
+                result = true;
+            }
```

Comment:
I think it does not prevent the ephemeral tx from setting less token amount than one that the contract call actually consumes. (I.e., the amount in `data` is less than `txData.tokenAmount`.)

Reply:
Thank you for your review, I've already fixed that point in my local. But now I'm running forge command in other repo using foundry-zksync. Let me check build and test in my local later.